### PR TITLE
feat: add i18n english translations for the pug files

### PR DESCRIPTION
**Pull request by kbrownlees**
Thanks for the great repository!

More than happy to take any feedback and improve this, feels like a good start to translate the initial landing pages.

### CODE REVIEW

**Review by rocwind — COMMENTED** [review]
Thanks for the PR! It looks great to me, just added a few comments regarding some details. I saw you put `cn` as the default locale, guess it's for keeping the previous behavior unchanged without explict config. I'd like to sugguest that we default to `en` instead:
1. `en` should have been used as default if we support the i18n at the first place. It would help if someone wants to translate the strings into other languages if we put `en` strings as default(key)
2. even though `cn` strings were hardcoded in pages before, I suppose changing to `en` should not cause too many troubles - given the fact that `code-push-cli` supports only `en` after all.
3. no matter which default locale we choose, it seems i18n will select one based on browser locale. It means `cn` users will likely still seeing `cn` strings.

File: src/app.ts [hunk 1]
```diff
@@ -23,10 +24,27 @@ app.use(
...
 // view engine setup
 app.set('views', path.join(__dirname, '../views'));
 app.set('view engine', 'pug');
 
+// translations
+
+app.use(i18n.init)
+
+app.use(function(req, res, next) {
+    // express helper for natively supported engines
```
[the remark this answers is not in the document]
Comment (kbrownlees) [reply src/app.ts:39]: Nice find! Will remove.

File: src/app.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import helmet from 'helmet';
 import { logger } from 'kv-logger';
 import { AppError, NotFound } from './core/app-error';
 import { config } from './core/config';
+import { i18n } from "./core/i81n"
```
[the remark this answers is not in the document]
Comment (kbrownlees) [reply src/app.ts:10]: Indeed there is, will rename.

File: src/core/i81n.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { I18n } from 'i18n'
+import path from "path"
+import { config } from "./config"
+
+export const i18n = new I18n({
+    locales: ['cn', 'en'],
+    directory: path.join(__dirname, '../../locales'),
+    defaultLocale: 'cn'
```
[the remark this answers is not in the document]
Comment (kbrownlees) [reply src/core/i81n.ts:8]: This is my take as well, I believe we can just leave the default as 'en' then have the init logic set the locale for each request based on the language header.

**Comment by kbrownlees** [comment]
> Thanks for the PR! It looks great to me, just added a few comments regarding some details. I saw you put `cn` as the default locale, guess it's for keeping the previous behavior unchanged without explict config. I'd like to sugguest that we default to `en` instead:
> 
>     1. `en` should have been used as default if we support the i18n at the first place. It would help if someone wants to translate the strings into other languages if we put `en` strings as default(key)
> 
>     2. even though `cn` strings were hardcoded in pages before, I suppose changing to `en` should not cause too many troubles - given the fact that `code-push-cli` supports only `en` after all.
> 
>     3. no matter which default locale we choose, it seems i18n will select one based on browser locale. It means `cn` users will likely still seeing `cn` strings.

Thanks for the quick response!

Having the default an 'en' would certainly be easier for me so I will update it. I was indeed trying to keep the previous behaviour.

I am also new to i18n so plenty of learning to do.

**Pull request merged**